### PR TITLE
Fix pill navigation when in Dashboard Sharing view-only context

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #5388

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Comment by tofumatt** [comment]
Just a note: this is working as-intended, but the final test to show the correct pill is being selected as the default is proving very difficult to write because the tests seem to always think the Traffic pill should be shown. I'm not sure why, but I'll take another look next week…

File: assets/js/components/DashboardNavigation/Navigation.js [hunk 1]
```diff
@@ -161,22 +204,31 @@ export default function DashboardNavigation() {
...
 
-		const chipID = initialHash;
+		let chipID = initialHash;
+
+		if ( ! isValidChipID( chipID ) ) {
+			chipID = defaultChipID;
+		}
+
 		setValue( ACTIVE_CONTEXT_ID, chipID );
+		setSelectedID( chipID );
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/DashboardNavigation/Navigation.js:225]: Tested and it works perfectly :)

**Comment by asvinb** [comment]
LGTM!

**Review by techanvil — APPROVED** [review]
LGTM! Nice one @tofumatt

**Pull request merged**